scripts/docker-compose-setup.sh

```diff
@@ -0,0 +1,60 @@
+#! /bin/sh
+
+# docker-compose-setup.sh
+# bring up local development environment via docker-compose
+#
+# More context: 
+# See https://github.com/broadinstitute/scp-ingest-pipeline#docker
+
+usage=$(
+cat <<EOF
+$0 [OPTION]
+-i   Set URL for GCR image; helpful if not using latest development
+-t   Set GitHub Vault token (e.g. ~/.github-token)
+-h   print this text
+EOF
+)
+
+GCR_IMAGE=""
+VAULT_TOKEN_PATH=""
+while getopts "i:t:h" OPTION; do
+case $OPTION in
+  i)
+    echo "### SETTING GCR IMAGE ###"
+    export GCR_IMAGE="$OPTARG"
+    ;;
+  t)
+    echo "### SETTING VAULT TOKEN ###"
+    VAULT_TOKEN_PATH="$OPTARG"
+    ;;
+  h)
+  	echo "$usage"
+  	exit 0
+  	;;
+  *)
+  	echo "unrecognized option"
+  	echo "$usage"
+  	exit 1
+  	;;
+  esac
+done
+
+if [[ $GCR_IMAGE = "" ]]; then
+  IMAGE_NAME="gcr.io/broad-singlecellportal-staging/scp-ingest-pipeline-development"
+  LATEST_TAG=$(gcloud container images list-tags ${IMAGE_NAME} --format='get(tags)' | head -n 1)
+  export GCR_IMAGE="${IMAGE_NAME}:${LATEST_TAG}"
+fi
+
+if [[ $VAULT_TOKEN_PATH = "" ]]; then
+  echo "Did not provide VAULT_TOKEN_PATH"
+  exit 1
+fi
+
+echo "### SETTING UP ENVIRONMENT ###"
+./scripts/ingest-local-setup.sh $VAULT_TOKEN_PATH
+
+docker pull $GCR_IMAGE
+
+# Run the "ingest" service (define in docker-compose-dev.yaml), 
+# then enter the Bash shell
+docker-compose -f docker-compose-dev.yaml run ingest bash
```

Comment:
Can someone walk me through how `docker-compose-dev.yaml` gets created? I've lost the thread on how this file fits in...

Reply:
The high-level explanation is that `docker-compose-dev.yaml` is a static representation of all the arguments that you would pass to `docker run`.  This allows you to define and bring up/down a service (or collection of services like we do in `single_cell_portal_core`) with a single call to `docker-compose`.  We then pair that with `docker-compose-setup.sh` to automate calling `vault` and pulling the correct Docker image beforehand so that everything is streamlined.

Those two files essentially replicate the following:

```
vault login [credentials_path] >| config/.ingest_env
source config/.ingest_env
docker pull [image name]
docker run [arguments] bash
```

Reply:
Thank you! I _thought_ this was probably where `config/.ingest_env` was being used but didn't want to assume...
and after `docker run` the process is tied up running the instance so you can't use this script to delete `config/.ingest_env`. That's too bad but makes sense.